lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/table/FileViewUtils.java

```diff
@@ -0,0 +1,290 @@
+package org.sagebionetworks.repo.model.dbo.dao.table;
+
+import static org.sagebionetworks.repo.model.dbo.dao.table.FileEntityFields.currentVersion;
+import static org.sagebionetworks.repo.model.dbo.dao.table.FileEntityFields.dataFileHandleId;
+import static org.sagebionetworks.repo.model.dbo.dao.table.FileEntityFields.id;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.COL_CURRENT_REV;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.COL_NODE_ID;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.COL_NODE_PARENT_ID;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.COL_NODE_TYPE;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.COL_REVISION_ANNOS_BLOB;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.COL_REVISION_NUMBER;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.COL_REVISION_OWNER_NODE;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.TABLE_REVISION;
+
+import java.io.IOException;
+import java.sql.Blob;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.sagebionetworks.repo.model.EntityType;
+import org.sagebionetworks.repo.model.NamedAnnotations;
+import org.sagebionetworks.repo.model.jdo.JDOSecondaryPropertyUtils;
+import org.sagebionetworks.repo.model.table.ColumnModel;
+import org.sagebionetworks.repo.model.table.ColumnType;
+import org.sagebionetworks.repo.model.table.Row;
+import org.sagebionetworks.util.ValidateArgument;
+
+/**
+ * Static utilities for FileViews.
+ *
+ */
+public class FileViewUtils {
+	
+	private static final String IDS_PARAM_NAME = "ids_param";
+
+	/**
+	 * Create the SQL need to build a view using the given column models.
+	 * 

```

Comment:
How the SQl will look like with parts that are required and parts that are optional

SELECT N.ID, N.CURRENT_VERSION <, OPTIONAL PRIMARY COLUMNS><, OPTIONAL ANNOTATION COLUMN>
FROM NODE N <, REVISION R>
WHERE N.TYPE = 'file'
AND N.PARENT_ID IN (<a given set of containers>)
<AND N.ID = R.OWNER_ID AND N.CURRENT_VERSION = R.VERSION_NUMBER>;